### DISCUSSION TITLE
Microsoft IDP reauthentication issue on Safari

### DISCUSSION THREAD

**Discussion by ML72** [keycloak/keycloak#21470] [Q&A]
We have a spring boot app that uses Keycloak for authentication, and the following sequence of actions causes an error when testing on Safari:
1. Sign in using Microsoft as the identity provider
2. Sign out of our app - which calls the Keycloak server's logout endpoint
3. Sign in again using Microsoft as the identity provider

Now, Keycloak will say that we are already signed in even though we are not; when we go back to our app, it shows that we are not signed in.

This issue *only* occurs on Safari (chrome and edge are fine), and *only* occurs with Microsoft IDP (Google and Apple are fine in any order of repeated authentication). Also, it is worth noting that if we sign out of our Microsoft account between steps 2 and 3, the issue does not seem to occur.

After debugging, it feels like the issue lies in our Keycloak template:
- In Azure AD logs, the authentication attempts all show as success
- When viewing the Keycloak client sessions, a new session is created for both steps 1 and 3

We have been trying to debug our template further, but have made little progress. At this point, we're not sure if this is a problem with the redirects and/or cookies on our Keycloak server, or if this is just a known issue with Safari & Microsoft. Any help/insight on this would be greatly appreciated!

**Comment by jonkoops** [comment]
You might want to try the latest [nightly version](https://github.com/keycloak/keycloak/releases/tag/nightly), we recently fixed some issues that were occurring in Safari  (see #21307).

**Reply by ML72** [reply]
Cool I'll take a look, thanks!
